src/classes/STG_SettingsManager_TEST.cls

```diff
@@ -1915,6 +1941,34 @@ private with sharing class STG_SettingsManager_TEST {
 
     }
 
+    /**
+    * @description Verifies that when Disable_Record_Data_Health_Checks__c is enabled,
+    *    all data record health checks should be disabled and have warning records created
+    */
+    @isTest
+    private static void shouldCreateWarningRecordsWhenDisabledRecordHealthCheckIsEnabled() {
+        STG_PanelHealthCheck_CTRL panel = new STG_PanelHealthCheck_CTRL();
+
+        STG_Panel.stgService.stgErr.Disable_Record_Data_Health_Checks__c = true;
+        
+        Test.startTest();
+        panel.runDetector();
+        Test.stopTest();
+
+        List<STG_PAnelHealthCheck_CTRL.DetectResult> results = panel.listDR;
+        Boolean hasDisabledHealthCheckWarningMessage = false;
+
+        for (STG_PAnelHealthCheck_CTRL.DetectResult result : results) {
+            if (result.strStatus == STG_PanelHealthCheck_CTRL.statusWarning && result.strName == System.Label.healthLabelDataBatchNotRun) {
+                hasDisabledHealthCheckWarningMessage = true;               
```

Comment:
this could be changed to hasDissabledHealthCheckWarningMessage = result.strStatus == STG(...)

Reply:
This will not work because there may be multiple results coming back and I do not want the boolean flag to be set to false because of other result